lib/matplotlib/backend_tools.py

```diff
@@ -750,6 +750,9 @@ def disable(self, event):
     def trigger(self, sender, event, data=None):
         self.toolmanager.get_tool(_views_positions).add_figure(self.figure)
         super().trigger(sender, event, data)
+        new_navigate_mode = self.name.upper() if self.toggled else None
+        for ax in self.figure.axes:
+            ax._navigate_mode = new_navigate_mode
```

Comment:
Seems like this should call `Axes.set_navigate_mode` instead of modifying the private attribute.

Reply:
oh yes, you are right. I don't know how I missed that.